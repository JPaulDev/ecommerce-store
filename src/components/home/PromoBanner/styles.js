import styled from 'styled-components';

export const Container = styled.div`
  background-color: #ebebeb;
  background-image: url(${({ background }) => background});
  background-position: calc(100% + 220px) 50%;
  background-repeat: no-repeat;
  background-size: auto 186px;
  border: 1px solid var(--color-grey-12);
  margin-top: 15px;
  min-height: 185px;
  overflow: hidden;
  padding: 15px;
  position: relative;

  @media screen and (min-width: 550px) {
    padding: 15px 25px;
  }

  @media screen and (min-width: 750px) {
    background-position: calc(100% + 345px) 50%;
    background-size: auto 291px;
    min-height: 250px;
    padding: 25px;
  }

  @media screen and (min-width: 850px) {
    background-position: center;
  }
`;

export const Heading = styled.h4`
  font-family: 'Uni Sans', sans-serif;
  font-size: var(--font-size-28);

  @media screen and (min-width: 750px) {
    font-size: var(--font-size-38);
  }
`;

export const Text = styled.p`
  font-family: 'Uni Sans Book', sans-serif;
  line-height: 1.3;
  margin: 4px 0 10px;
  width: 45%;

  @media screen and (min-width: 750px) {
    font-size: var(--font-size-20);
    width: 35%;
  }
`;

export const LogoWrapper = styled.div`
  position: absolute;
  right: 25px;
  top: 25px;

  @media screen and (max-width: 849px) {
    display: none;
  }
`;

export const Wrapper = styled.div`
  bottom: -110px;
  position: absolute;
  right: -70px;

  @media screen and (min-width: 750px) {
    right: 10px;
  }
`;
