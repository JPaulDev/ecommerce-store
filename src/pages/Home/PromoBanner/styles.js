import styled from 'styled-components';

export const Container = styled.div`
  background-color: #ebebeb;
  background-image: url(${({ background }) => background});
  background-position: calc(100% + 220px) 50%;
  background-repeat: no-repeat;
  background-size: auto 186px;
  border: 1px solid var(--color-grey-9);
  margin-top: 15px;
  min-height: 185px;
  overflow: hidden;
  padding: 15px;
  position: relative;

  @media screen and (min-width: 550px) {
    padding: 15px 25px;
  }

  @media screen and (min-width: 750px) {
    background-position: calc(100% + 345px) 50%;
    background-size: auto 291px;
    min-height: 250px;
    padding: 25px;
  }

  @media screen and (min-width: 850px) {
    background-position: center;
  }
`;

export const Title = styled.h4`
  font-family: 'Uni Sans', sans-serif;
  font-size: 1.75rem;

  @media screen and (min-width: 750px) {
    font-size: 2.6rem;
  }
`;

export const Text = styled.p`
  font-family: 'Uni Sans Book', sans-serif;
  font-size: 1.06rem;
  line-height: 1.3;
  margin: 4px 0 10px;
  width: 45%;

  @media screen and (min-width: 750px) {
    font-size: 1.2rem;
    width: 35%;
  }
`;

export const Logo = styled.img`
  height: 73px;
  position: absolute;
  right: 25px;
  top: 25px;
  width: 93px;
`;

export const BlueDetail = styled.img`
  bottom: -105px;
  position: absolute;
  right: -70px;

  @media screen and (min-width: 750px) {
    bottom: -100px;
    right: 10px;
  }
`;
