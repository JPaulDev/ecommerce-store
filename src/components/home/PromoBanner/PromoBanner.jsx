import Image from 'next/future/image';
import nvidiaImage from '../../../../public/images/promo-banner/rtx-studio.webp';
import nvidiaLogo from '../../../../public/images/promo-banner/nvidia-logo.webp';
import blueDetail from '../../../../public/images/promo-banner/blue-detail.webp';
import useMediaQuery from '../../../hooks/useMediaQuery';
import { LinkButton, LinkChevron } from '../../common';
import * as Styled from './styles';

export default function PromoBanner() {
  const isMatch = useMediaQuery('(min-width: 530px)');
  const linkText = 'Find Out More';

  return (
    <Styled.Container background={nvidiaImage.src}>
      <Styled.Title>RTX Studio</Styled.Title>
      <Styled.Text>
        The best laptops and desktops for creating anywhere.
      </Styled.Text>
      {isMatch ? (
        <LinkButton href="/" linkText={linkText} padding="12px 14px" />
      ) : (
        <LinkChevron
          href="/"
          linkText={linkText}
          fontSize="0.8rem"
          iconSize={18}
        />
      )}
      <Styled.LogoWrapper>
        <Image src={nvidiaLogo} alt="" />
      </Styled.LogoWrapper>
      <Styled.Wrapper>
        <Image src={blueDetail} alt="" />
      </Styled.Wrapper>
    </Styled.Container>
  );
}
