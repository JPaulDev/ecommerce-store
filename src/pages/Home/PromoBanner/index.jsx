import useMediaQuery from '../../../hooks/useMediaQuery';
import nvidiaImage from '../../../assets/images/promo-banner/rtx-studio.webp';
import nvidiaLogo from '../../../assets/images/promo-banner/nvidia-logo.webp';
import blueDetail from '../../../assets/images/promo-banner/blue-detail.webp';
import LinkButtonGrey from '../../../components/LinkButtonGrey';
import LinkWithChevron from '../../../components/LinkWithChevron';
import * as Styled from './styles';

export default function AdvertBanner() {
  const isMatch530 = useMediaQuery('(min-width: 530px)');
  const isMatch850 = useMediaQuery('(min-width: 850px)');
  const linkText = 'Find Out More';

  return (
    <Styled.Container background={nvidiaImage}>
      <Styled.Title>RTX Studio</Styled.Title>
      <Styled.Text>
        The best laptops and desktops for creating anywhere.
      </Styled.Text>
      {isMatch530 ? (
        <LinkButtonGrey href="" padding="12px 14px">
          {linkText}
        </LinkButtonGrey>
      ) : (
        <LinkWithChevron
          url=""
          text={linkText}
          fontSize="0.75rem"
          iconSize="18px"
        />
      )}
      {isMatch850 && <Styled.Logo src={nvidiaLogo} alt="" />}
      <Styled.BlueDetail src={blueDetail} alt="" />
    </Styled.Container>
  );
}
