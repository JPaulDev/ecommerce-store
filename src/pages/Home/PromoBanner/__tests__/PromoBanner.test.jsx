import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import PromoBanner from '../index';

describe('PromoBanner component', () => {
  it('renders a promotional banner about RTX studio', () => {
    const tree = renderer.create(<PromoBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('PromoBanner component at a screen width of 850px and above', () => {
  it('renders a logo image', () => {
    window.resizeTo(850, 768);
    render(<PromoBanner />);

    expect(screen.getByTestId('logo')).toBeInTheDocument();
  });
});

describe('PromoBanner component at a screen width of 849px and below', () => {
  it('does not render a logo image', () => {
    window.resizeTo(849, 768);
    render(<PromoBanner />);

    expect(screen.queryByTestId('logo')).toBeNull();
  });
});

describe('PromoBanner component at a screen width of 530px and above', () => {
  beforeEach(() => {
    window.resizeTo(530, 768);
    render(<PromoBanner />);
  });

  it('renders a link button', () => {
    expect(screen.getByTestId('link-button')).toBeInTheDocument();
  });

  it('does not render a link with a chevron', () => {
    expect(screen.queryByTestId('link-chevron')).toBeNull();
  });
});

describe('PromoBanner component at a screen width of 529px and below', () => {
  beforeEach(() => {
    window.resizeTo(529, 768);
    render(<PromoBanner />);
  });

  it('renders a link with a chevron', () => {
    expect(screen.getByTestId('link-chevron')).toBeInTheDocument();
  });

  it('does not render a link button', () => {
    expect(screen.queryByTestId('link-button')).toBeNull();
  });
});
