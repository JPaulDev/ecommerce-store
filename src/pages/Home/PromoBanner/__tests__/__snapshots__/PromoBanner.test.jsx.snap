// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PromoBanner component renders a promotional banner about RTX studio 1`] = `
<div
  className="sc-eCYdqJ cDKfSV"
>
  <h4
    className="sc-jSMfEi jEiboe"
  >
    RTX Studio
  </h4>
  <p
    className="sc-gKXOVf krPcbR"
  >
    The best laptops and desktops for creating anywhere.
  </p>
  <a
    className="sc-bczRLJ diqCvw"
    data-testid="link-button"
    href=""
  >
    Find Out More
  </a>
  <img
    alt=""
    className="sc-iBkjds ibOmqv"
    data-testid="logo"
    src="nvidia-logo.webp"
  />
  <img
    alt=""
    className="sc-ftvSup iSGTUF"
    src="blue-detail.webp"
  />
</div>
`;
